### PR TITLE
KAFKA-6317: Maven artifact for kafka should not depend on log4j

### PR DESCRIPTION
**Pull request by ijuma**
It should only depend on slf4j-api (like kafka-clients). The
release tarball still includes log4j and slf4j-log4j12.

Manually verified that there are no duplicate dependencies
in the release tarball and `./gradlew core:dependencies`
looks good.

### Committer Checklist (excluded from commit message)
- [x] Verify design and implementation 
- [x] Verify test coverage and CI build status
- [x] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by ijuma** [comment]
cc @guozhangwang @rhauch @ewencp @viktorsomogyi

File: docs/upgrade.html [hunk 1]
```diff
@@ -19,6 +19,68 @@
...
+</ul>
+
+<h5><a id="upgrade_110_new_protocols" href="#upgrade_110_new_protocols">New Protocol Versions</a></h5>
+<ul></ul>
+
+<h5><a id="upgrade_110_streams" href="#upgrade_110_streams">Upgrading a 1.1.0 Kafka Streams Application</a></h5>
+<ul>
+    <li> Upgrading your Streams application from 1.0.0 to 1.1.0 does not require a broker upgrade.
+        A Kafka Streams 1.1.0 application can connect to 1.0, 0.11.0, 0.10.2 and 0.10.1 brokers (it is not possible to connect to 0.10.0 brokers though). </li>
+    <li> If you are monitoring on streams metrics, you will need make some changes to the metrics names in your reporting and monitoring code, because the metrics sensor hierarchy was changed. </li>
```
Comment (guozhangwang) [inline docs/upgrade.html]: This list seems to be only for 1.0.0 ?

File: docs/upgrade.html [hunk 2]
```diff
@@ -19,6 +19,68 @@
 
 <script id="upgrade-template" type="text/x-handlebars-template">
 
+<h4><a id="upgrade_1_1_0" href="#upgrade_1_1_0">Upgrading from 0.8.x, 0.9.x, 0.10.0.x, 0.10.1.x, 0.10.2.x, 0.11.0.x or 1.0.x to 1.1.x</a></h4>
+<p>Kafka 1.1.0 introduces wire protocol changes. By following the recommended rolling upgrade plan below,
```
Comment (guozhangwang) [inline docs/upgrade.html:23]: What wire protocol changes were introduced in 1.1.0?

File: docs/upgrade.html (hunk 1 above)
Comment (ijuma) [reply docs/upgrade.html]: Copy and paste error, will fix.

File: docs/upgrade.html (hunk 2 above)
Comment (ijuma) [reply docs/upgrade.html:23]: None, but we are planning to do so.

File: docs/upgrade.html (hunk 1 above)
Comment (ijuma) [reply docs/upgrade.html]: Fixed.

File: build.gradle [hunk 3]
```diff
@@ -945,7 +958,7 @@ project(':streams:examples') {
   dependencies {
     compile project(':streams')
     compile project(':connect:json')  // this dependency should be removed after we unify data API
-    compile libs.slf4jlog4j           // this dependency should be removed after KIP-4
+    compile libs.slf4jlog4j
```
Comment (ijuma) [inline build.gradle:961]: Removed the comment because I think it was wrong.

File: build.gradle [hunk 4]
```diff
@@ -595,7 +603,6 @@ project(':core') {
     compile.exclude module: 'jmxri'
     compile.exclude module: 'jmxtools'
     compile.exclude module: 'mail'
-    compile.exclude module: 'netty'
```
Comment (viktorsomogyi) [inline build.gradle:598]: Why was this included?

Comment (ijuma) [reply build.gradle:598]: I moved it to the zookeeper exclusions.

Comment (viktorsomogyi) [reply build.gradle:598]: Ah ok.

**Comment by viktorsomogyi** [comment]
@ijuma, a general question: how and from where does the log4j jars get copied into the release targz? Looking at `core:dependencies` it really adds it as a `compileOnly` dependency but in `build.gradle` I don't see any log4j jars copied when creating the release tarball (and I guess the `testRuntime` copies don't count :) ) - but I might have missed something.

**Comment by ijuma** [comment]
@viktorsomogyi much of the PR is dedicated to that, see `copyDependantLibs`.

**Comment by viktorsomogyi** [comment]
@ijuma my bad, I was under the impression the whole time that those copy INTO testRuntime, not FROM. Everything is fine now, sorry for the confusion.

**Review by viktorsomogyi — APPROVED** [review]
I have checked out the fix, ran the `core:dependencies` task and build the release tgz and the change seems to be correct.

**Comment by ijuma** [comment]
@rajinisivaram, maybe you can review this.

**Review by rajinisivaram — APPROVED** [review]
@ijuma Thanks for the PR, LGTM
